File: plant.py
from organism import Organism
from Stale import Stale

class Plant(Organism):
    def Collision(self, attacked):
        self.world.GetOrganisms().remove(self, attacked)
        # !!! NOTIFICATIONS WHEN ANIMAL IS DELETED

    def Action(self):
        # EMPTY PLACE FOR BREEDING
        place = self.EmptyPlace()
        tmp_cords = self.GetCords()

        # IF THERE IS ANY EMPTY PLACE
        if place != -1:
            if place == Stale.LEFT:
                self.New(self.getName(), self.world, tmp_cords.x - 1, tmp_cords.y)
            elif place == Stale.RIGHT:
                self.New(self.getName(), self.world, tmp_cords.x + 1, tmp_cords.y)
            elif place == Stale.UP:
                self.New(self.getName(), self.world, tmp_cords.x, tmp_cords.y - 1)
            elif place == Stale.DOWN:
                self.New(self.getName(), self.world, tmp_cords.x, tmp_cords.y + 1)

    def Drawing(self):
        pass

    def getName(self):
        return ""

File: Plant.java
public class Plant extends Organism{
    @Override
    public  void Collision(Organism attacked){
        world.GetOrganisms().remove(this,attacked);
        //!!!NOTIFICATIONS WHEN ANIMAL IS DELETED
    }
    @Override
    public void Action(){
        //EMPTY PLACE FOR BREADING
        int place = EmptyPLace();
        Polozenie tmp_cords = GetCords();

        //IF THERE IS ANY EMPTY PLACE
        if(place != -1){
            if(place==Stale.LEFT){
                New(getName(), world, tmp_cords.x - 1, tmp_cords.y);
            }
            else if(place==Stale.RIGHT){
                New(getName(), world, tmp_cords.x + 1, tmp_cords.y);
            }
            else if(place==Stale.UP){
                New(getName(), world, tmp_cords.x, tmp_cords.y - 1);
            }
            else if(place==Stale.DOWN){
                New(getName(), world, tmp_cords.x, tmp_cords.y + 1);
            }
        }
    }
    @Override
    public  void Drawing(){}
    @Override
    public String getName(){
        return "";
    }
}
